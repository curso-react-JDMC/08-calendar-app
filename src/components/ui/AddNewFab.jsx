import React from "react";
import { useDispatch } from "react-redux";
import { calendarEventClearActive } from "../../redux/actions/calendar";
import { uiOpenModal } from "../../redux/actions/ui";

export const AddNewFab = () => {
  const dispatch = useDispatch();

  const handleAddNew = () => {
    dispatch(uiOpenModal());
    dispatch(calendarEventClearActive())
  }
  return (
    <button className="btn btn-primary fab" onClick={handleAddNew}>
      <i className="fas fa-plus"></i>
    </button>
  );
};
