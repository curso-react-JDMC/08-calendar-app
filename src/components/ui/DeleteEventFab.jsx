import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { calendarEventDelete } from "../../redux/actions/calendar";

export const DeleteEventFab = () => {
  const dispatch = useDispatch();
  const { activeEvent } = useSelector((state) => state.calendar);


    const handleDelete = () =>{
        dispatch(calendarEventDelete())
    }

  return (
    <>
      {activeEvent && (
        <button className="btn btn-danger fab-danger" onClick={ handleDelete }>
          <i className="fas fa-trash"></i>
          <span> Borrar Evento </span>
        </button>
      )}
    </>
  );
};
