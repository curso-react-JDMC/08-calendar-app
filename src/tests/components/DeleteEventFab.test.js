import { DeleteEventFab } from "../../components/ui/DeleteEventFab";
import configureStore from "redux-mock-store"; //ES6 modules
import thunk from "redux-thunk";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import { render, screen } from "@testing-library/react";
import * as userEvent from '@testing-library/user-event';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initialState = { activeEvent: false };
const store = mockStore(initialState);
store.dispatch = jest.fn();

jest.mock("react-redux", () => ({
  useSelector: () => ({ activeEvent: true }),
  useDispatch: jest.fn(),
}));

// const wrapper = mount(
//   <Provider store={store}>
//     <DeleteEventFab/>
//   </Provider>
// );

describe("Pruebas en <DeleteEventFab/>", () => {
  test("debe mostrarse correctamente", () => {
    //expect(wrapper).toMatchSnapshot();
    render(
        <DeleteEventFab />
    );
    expect(render).toMatchSnapshot();
  });
//   test('debe llamar eventStartDelete', async() => { 
//     const user = userEvent.setup()
//     render(
//         <DeleteEventFab />
//     );
//     expect(screen.getByRole('button')).toHaveTextContent("Borrar Evento");
//     await user.click(screen.getByRole('button'));
//   })
});
