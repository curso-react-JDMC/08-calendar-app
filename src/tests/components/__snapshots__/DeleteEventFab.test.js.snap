// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DeleteEventFab/> debe mostrarse correctamente 1`] = `[Function]`;
